node/src/actors/session/handlers.rs

```diff
@@ -832,7 +832,10 @@ fn handshake_version(
     // Check timestamp drift
     let received_ts = command_version.timestamp;
     let max_ts_diff = session.config.connections.handshake_max_ts_diff;
-    if max_ts_diff != 0 && (current_ts - received_ts).abs() > max_ts_diff {
+    if max_ts_diff != 0
+        && (current_ts.saturating_sub(max_ts_diff)..=current_ts.saturating_add(max_ts_diff))
```

Comment:
are you sure that it follows the same logic?

Reply:
No, but I am sure that it does not panic. I can extract this line to a function and add tests, because after this change it is not very clear.

Reply:
Oh you are right, it should be the inverse condition.